components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -1325,7 +1325,8 @@ static JsonNode getResponseCacheConfig(String responseCache, int cacheTimeout) {
     public static String preProcess(String swaggerContent) throws APIManagementException {
         //Load required properties from swagger to the API
         APIDefinition apiDefinition = getOASParser(swaggerContent);
-        return apiDefinition.processOtherSchemeScopes(swaggerContent);
+        swaggerContent = apiDefinition.processOtherSchemeScopes(swaggerContent);
+        return apiDefinition.processLegacyScopes(swaggerContent);
```

Comment:
shouldn't processLegacyScopes execute before the processOtherSchemeScopes?

Reply:
processLegacyScopes will add the default schema to SecuritySchemes and since we now skip the processOtherSchemeScopes if the default schema already exists in the definition we have to call processLegacyScopes after calling processOtherSchemeScopes.
If we are going to call processLegacyScopes before calling processOtherSchemeScopes we have to copy the legacy scopes to some oauth2 scheme other than 'default' and let them be copied to 'default' scheme when processOtherSchemeScopes is executed.

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/9012/commits/b188ce5404a9c6d4634d3c3ad7d44eba254e4437

Reply:
Yes. Got it.